go/mysql/mysql56_gtid_set.go

```diff
@@ -155,7 +155,7 @@ func (set Mysql56GTIDSet) String() string {
 		if i != 0 {
 			buf.WriteByte(',')
 		}
-		buf.WriteString(sid.String())
+		buf.Write(sid[:])
```

Comment:
I don't think this is equivalent? It also shows in the `endtoend` test suite:

```
vttablet: rpc error: code = Internal desc = could not decode position: invalid MySQL 5.6 GTID set ("\xd1\xf8\b\x8dl.\x11\xed\x8ao\x00\r:;\xcep:1-144"): invalid MySQL 5.6 SID "\xd1\xf8\b\x8dl.\x11\xed\x8ao\x00\r"
[1090](https://github.com/vitessio/vitess/actions/runs/3543219398/jobs/5949479892#step:12:1091)
```

This writes the bytes directly, but `sid.String()` converts it to a hex representation for the string which other places depend on.

Reply:
@dbussink 👋 yeah, I'm seeing the same test failures now. I'm not sure how I missed that 🤦 

I'll close this PR for now and look further into it